syft/pkg/cataloger/golang/parse_go_mod.go

```diff
@@ -151,3 +182,353 @@ func parseGoSumFile(resolver file.Resolver, reader file.LocationReadCloser) (map
 
 	return out, nil
 }
+
+// loadPackages uses golang.org/x/tools/go/packages to get dependency information.
+func (c *goModCataloger) loadPackages(modDir string) (pkgs map[string][]pkgInfo, modules map[string]*packages.Module, dependencies map[string][]string) {
+	cfg := &packages.Config{
+		Mode:  packages.NeedModule | packages.NeedName | packages.NeedFiles | packages.NeedDeps | packages.NeedImports,
```

Comment:
needs comments on why these options are selected and the performance impact of expanding this selection